test/steps/infinispan.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2020 Red Hat, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package steps
+
+import (
+	"fmt"
+
+	infinispan "github.com/infinispan/infinispan-operator/pkg/apis/infinispan/v1"
+
+	"github.com/cucumber/godog"
+	"github.com/cucumber/messages-go/v10"
+	"github.com/kiegroup/kogito-cloud-operator/test/framework"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	//DataTable first column
+	securityKey = "security"
+
+	//DataTable second column
+	endpointSecretNameKey = "EndpointSecretName"
+)
+
+func registerInfinispanSteps(s *godog.Suite, data *Data) {
+	s.Step(`^Infinispan instance "([^"]*)" is deployed with configuration:$`, data.infinispanInstanceIsDeployedWithConfiguration)
+
+	// Secret steps
+	s.Step(`^Infinispan Secret "([^"]*)" is created with credentials:$`, data.infinispanSecretIsCreatedWithCredentials)
+}
+
+func (data *Data) infinispanInstanceIsDeployedWithConfiguration(name string, table *messages.PickleStepArgument_PickleTable) error {
+	infinispan := &infinispan.Infinispan{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: data.Namespace,
+			Name:      name,
+		},
+		Spec: infinispan.InfinispanSpec{
+			Replicas: 1,
+		},
+	}
+
+	if err := configureInfinispanFromTable(table, infinispan); err != nil {
+		return err
+	}
+
+	if err := framework.DeployInfinispanInstance(data.Namespace, infinispan); err != nil {
+		return err
+	}
+
+	return framework.WaitForPods(data.Namespace, "app", "infinispan-pod", 1, 3)
```

Comment:
can we rename that method to `WaitForPodsWithLabel` ? It took me some time to understand what was done here

Reply:
sure, will do